.buildkite/scripts/bintray_upload_macos.sh

```diff
@@ -24,7 +24,7 @@ sudo hab pkg install \
      core/hab-bintray-publish
 
 echo "--- :buildkite: Retrieving macOS core/hab artifact"
-hab_artifact=$(buildkite-agent meta-data get "hab-artifact-macos")
+hab_artifact=$(buildkite-agent meta-data get "hab-artifact-x86_64-darwin")
```

Comment:
Wonder if we can get a bit more consistency, and maybe be able to abstract some of these metadata calls into functions, if we require a package target environment variable to be set everywhere, and then take advantage of that here, rather than hardcoding?

Reply:
I started to look at this because I think it's the right idea, but I've hit a bit of a snag.  We use the `meta-data get` for both the current $pkg_target, but in the case of bintray_publish for _all_ $pkg_targets. 

Creating a function that always requires the target be passed doesn't get us much (anything?) from requiring a Target environment variable. Having the target be optional I'd argue is worse because it would be pretty easy to miss a spot where it's necessary and hard to debug in the pipeline.  

We could split out bintray_publish to act only on a single target, and have a step for each target... I think this is what I'll try. Refactoring all scripts to act on a single target where appropriate. It means we'd need N workers to run each curl rather than 1 worker to run N curls. I think the failure case of one not working isn't any worse than current state either.

Reply:
I may have been overly prescriptive... the function doesn't need to operate on the specific environment variable, but we could take advantage of the environment variable and pass it in as an argument.

One thing this would allow us to do is hide the nature of the _keys_ behind a function. For example, we do that here: https://github.com/habitat-sh/habitat/blob/58f19cf1c2d756f4e2c8e2495a40fef4fd3c3f0d/.buildkite/scripts/shared.sh#L47-L61

Every place that needs to operate on the metadata for a "fake release" doesn't need to know what the metadata key is (or that it's even stored in Buildkite metadata). It feels like we're getting to the point where we're making more and more use of the metadata feature, to the point where it could become confusing and error prone to be replicating all this logic across the various scripts.

Then here, we'd just have a list of targets we care about (not even environment variables, necessarily) and we'd just iterate through them, operating on each appropriate package in turn.